### PR TITLE
SNOW-1445412: Add support for Series/DataFrame.at/iat

### PR DESCRIPTION
**Pull request by sfc-gh-helmeleegy**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1445412, SNOW-1445414, SNOW-1445417, SNOW-1445418

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

   Add support for Series/DataFrame.at/iat.

### CODE REVIEW

File: docs/source/modin/supported/series_supported.rst [hunk 1]
```diff
@@ -19,7 +19,7 @@ Attributes
 +-----------------------------+---------------------------------+----------------------------------------------------+
 | ``array``                   | N                               |                                                    |
 +-----------------------------+---------------------------------+----------------------------------------------------+
-| ``at``                      | N                               |                                                    |
+| ``at``                      | P                               | ``N`` for set with MultiIndex                      |
```
Comment (sfc-gh-nkrishna) [inline docs/source/modin/supported/series_supported.rst:22]: You missed changing the `N` to `Y` for `Series.iat`

File: src/snowflake/snowpark/modin/pandas/base.py [hunk 2]
```diff
@@ -1137,16 +1137,15 @@ def astype(
...
 
-    @base_not_implemented()
     @property
     def at(self, axis=None):  # noqa: PR01, RT01, D200
         """
         Get a single value for a row/column label pair.
         """
         # TODO: SNOW-1119855: Modin upgrade - modin.pandas.base.BasePandasDataset
-        from .indexing import _LocIndexer
+        from .indexing import _AtIndexer
```
Comment (sfc-gh-nkrishna) [inline src/snowflake/snowpark/modin/pandas/base.py:1146]: This does deviate from Modin, but I think this implementation with `_AtIndexer` and the error checking is much cleaner

**Review by sfc-gh-nkrishna — APPROVED** [review]
Nice work, approving! Please add docs change for `Series.iat`

File: docs/source/modin/supported/series_supported.rst (hunk 1 above)
Comment (sfc-gh-helmeleegy) [reply docs/source/modin/supported/series_supported.rst:22]: Done.

**Review by sfc-gh-dpetersohn — APPROVED** [review]
LGTM!

File: src/snowflake/snowpark/modin/pandas/base.py (hunk 2 above)
Comment (sfc-gh-dpetersohn) [reply src/snowflake/snowpark/modin/pandas/base.py:1146]: I agree

**Pull request merged**